agent/config/builder.go

```diff
@@ -1290,6 +1290,10 @@ func (b *builder) validate(rt RuntimeConfig) error {
 			"1 and 63 bytes.", rt.NodeName)
 	}
 
+	if err := rt.StructLocality().Validate(); err != nil {
+		return fmt.Errorf("'locality.zone' cannot be set without 'locality.region'")
```

Comment:
I included the top level field name here since I thought it would be confusing for us to return errors that look like they are for a top level field, but they are actually for a field nested under `locality`. I also added the single quotes for consistency with other config validation errors.

What do you think?

Reply:
I would say we should add to the error passed by validate, then if there are other errors in the future we leave it to validate to tell us what the error is, something like:

`return fmt.Errorf("locality is invalid: %s", err.Error())`